File: algorithms/[415]字符串相加/solution.py
# topics = ["字符串"]


class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        n = len(num1) - 1
        m = len(num2) - 1
        res = ''
        carry = 0

        while n >= 0 and m >= 0:
            sum = int(num1[n]) + int(num2[m]) + carry
            carry = sum // 10
            res = str(sum % 10) + res
            n -= 1
            m -= 1

        while n >= 0:
            sum = int(num1[n]) + carry
            carry = sum // 10
            res = str(sum % 10) + res
            n -= 1

        while m >= 0:
            sum = int(num2[m]) + carry
            carry = sum // 10
            res = str(sum % 10) + res
            m -= 1

        return str(carry) + res if carry else res

File: algorithms/[415]字符串相加/solution.ts
// topics = ["字符串"]

function addStrings(num1: string, num2: string): string {
  let n = num1.length - 1;
  let m = num2.length - 1;
  let res = '';
  let carry = 0;

  while (n >= 0 || m >= 0) {
    const first = n >= 0 ? parseInt(num1[n--]) : 0;
    const second = m >= 0 ? parseInt(num2[m--]) : 0;
    const val = first + second + carry;
    carry = Math.floor(val / 10);
    res = (val % 10) + res;
  }
  return carry ? carry + res : res;
}
